src/commands/infra/init.ts

```diff
@@ -0,0 +1,94 @@
+import child_process from "child_process";
+import commander from "commander";
+import emoji from "node-emoji";
+import { promisify } from "util";
+import { logger } from "../../logger";
+
+/**
+ * Adds the init command to the commander command object
+ *
+ * @param command Commander command object to decorate
+ */
+
+// Check for versions for each executable
+
+let binaries: string[] = ["terraform", "git", "az", "helm"];
+let envVar: string[] = [
+  "ARM_SUBSCRIPTION_ID",
+  "ARM_CLIENT_ID",
+  "ARM_CLIENT_SECRET",
+  "ARM_TENANT_ID"
+];
+
+export const initCommand = (command: commander.Command): void => {
+  command
+    .command("init")
+    .alias("i")
+    .description(
+      "Initialize will verify that all infrastructure deployment prerequisites have been correctly installed."
+    )
+    .action(async opts => {
+      try {
+        if (await validatePrereqs(binaries)) {
+          if (await validateAzure()) {
+            await validateEnvVariables(envVar);
+          }
+        }
+      } catch (err) {
+        logger.error(`Error validating init prerequisites`);
+        logger.error(err);
+      }
+    });
+};
+
+export const validatePrereqs = async (
+  executables: string[]
+): Promise<boolean> => {
+  // Validate executables in PATH
+  for (let i of executables) {
+    try {
+      await promisify(child_process.exec)("which " + i);
+    } catch (err) {
+      logger.error(
+        emoji.emojify(":no_entry_sign: '" + i + "'" + " not installed")
+      );
+      return false;
+    }
+  }
+  logger.info(
+    emoji.emojify("Installation of Prerequisites verified: :white_check_mark:")
+  );
+  return true;
+};
+
+export const validateAzure = async (): Promise<boolean> => {
```

Comment:
This method dependences on `validatePrereqs` being called first with `az` as one of the inputs. Because of this tight dependency can you see use combining these methods somehow?

Reply:
When you say "this method", did you mean the method `validateAzure`? If so, the answer is no, because we want to enforce a "fail fast" scenario. In a situation where Azure CLI is NOT installed, fail, before running any `az` command.